google-cloud-spanner/src/main/java/com/google/cloud/spanner/InstanceInfo.java

```diff
@@ -87,21 +91,31 @@ Builder setCreateTime(Timestamp createTime) {
     }
 
     /**
-     * Sets the number of nodes for the instance. Exactly one of processing units or node count must
-     * be set when creating a new instance.
+     * Sets the number of nodes for the instance. Exactly one of processing units, node count or
+     * autoscaling config must be set when creating a new instance.
      */
     public abstract Builder setNodeCount(int nodeCount);
 
     /**
-     * Sets the number of processing units for the instance. Exactly one of processing units or node
-     * count must be set when creating a new instance. Processing units must be between 1 and 999
-     * (inclusive) when creating a new instance with node count = 0. Processing units from 1000 and
-     * up must always be a multiple of 1000 (that is equal to an integer number of nodes).
+     * Sets the number of processing units for the instance. Exactly one of processing units, node
+     * count, or autoscaling config must be set when creating a new instance. Processing units must
+     * be between 1 and 999 (inclusive) when creating a new instance with node count = 0. Processing
+     * units from 1000 and up must always be a multiple of 1000 (that is equal to an integer number
+     * of nodes).
      */
     public Builder setProcessingUnits(int processingUnits) {
       throw new UnsupportedOperationException("Unimplemented");
     }
 
+    /**
+     * Sets the autoscaling config for the instance, which will enable the autoscaling for this
+     * instance. Exactly one of processing units, node count, or autoscaling config must be set when
+     * creating a new instance.
+     */
+    public Builder setAutoscalingConfig(AutoscalingConfig autoscalingConfig) {
```

Comment:
This method can be made abstract similar to `setNodeCount`

Reply:
Ah I see, updated. Should the `setProcessingUnits` also be abstract then?